flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java

```diff
@@ -126,10 +131,12 @@ public void deserialize(SourceRecord record, Collector<RowData> out) throws Exce
             delete.setRowKind(RowKind.DELETE);
             emit(record, delete, out);
         } else {
-            GenericRowData before = extractBeforeRow(value, valueSchema);
-            validator.validate(before, RowKind.UPDATE_BEFORE);
-            before.setRowKind(RowKind.UPDATE_BEFORE);
-            emit(record, before, out);
+            if (changelogMode == DebeziumChangelogMode.ALL) {
```

Comment:
minor：We can build runtime filter to deal `update_before` when construct the `DeserializeSchema` instead of judge it per record.